### PR TITLE
Treat intermediate columns specially.

### PR DESCRIPTION
**Pull request by chriseth**
This is needed because in the future, the values of intermediate columns need to be "simple expressions".

### CODE REVIEW

File: backend/src/pilstark/json_exporter/mod.rs [hunk 1]
```diff
@@ -55,7 +55,7 @@ pub fn export<T: FieldElement>(analyzed: &Analyzed<T>) -> PIL {
                                 exporter.intermediate_poly_expression_ids[&poly.id] as usize
                             );
                         } else {
-                            panic!("Expected single value");
+                            panic!("Expected single intermediate value");
```
Comment (leonardoalt) [inline backend/src/pilstark/json_exporter/mod.rs:58]: `single value` sounds a bit misleading imo, maybe `single intermediate expression`?

File: ast/src/analyzed/mod.rs [hunk 2]
```diff
@@ -107,7 +121,7 @@ impl<T> Analyzed<T> {
             // for each kind.
             self.committed_polys_in_source_order(),
             self.constant_polys_in_source_order(),
-            self.intermediate_polys_in_source_order(),
+            //self.intermediate_polys_in_source_order(),
```
Comment (leonardoalt) [inline ast/src/analyzed/mod.rs:124]: remove?

File: ast/src/analyzed/display.rs [hunk 3]
```diff
@@ -28,37 +28,44 @@ impl<T: Display> Display for Analyzed<T> {
...
+                                write!(f, "    let {name}")?;
+                                if let Some(value) = definition {
+                                    write!(f, "{value}")?
+                                }
+                                writeln!(f, ";")?
                             }
-                            writeln!(f, ";")?
                         }
+                    } else if let Some((symbol, definition)) = self.intermediate_columns.get(name) {
+                        let (name, _) = update_namespace(name, symbol.degree, f)?;
```
Comment (leonardoalt) [inline ast/src/analyzed/display.rs:64]: why does it need to "update namespace" and what does that mean?

Comment (chriseth) [reply ast/src/analyzed/display.rs:64]: At this stage ("analyzed"), we eliminated the concept of namespaces. All symbols have fully qualified names.

For various reasons, we need to re-introduce namespaces again for printing, that's what "update_namespace" does: If the namespaces changed, it emits a `namespace XYZ(n):` line and updates the current namespace.

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**